src/aiperf/common/mixins/buffered_jsonl_writer_mixin.py

```
@@ -135,26 +167,80 @@ async def _flush_buffer(self, buffer_to_flush: list[bytes]) -> None:
+                await asyncio.sleep(self._flush_interval)
+                if not self._buffer:
+                    continue
+                buffer_to_flush = self._buffer
+                self._buffer = []
+                # Shield the flush so a cancel (from _close_file during
+                # teardown) can't interrupt an in-flight write and silently
+                # drop the records we already pulled out of self._buffer.
+                await asyncio.shield(self._flush_buffer(buffer_to_flush))
+            except asyncio.CancelledError:
+                return
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)